lib/puppet-validator.rb

```diff
@@ -73,19 +75,59 @@ def initialize(app=nil)
     erb :index
   end
 
+  get '/load/*' do
+    location = params[:splat].first.sub(/(https?:)\/\/?/, '\1//')
+    logger.info "Loading code from: #{location}"
+
+    uri = location.downcase.start_with?('http') ? URI.parse(location) : URI.parse("https://#{location}")
+
+    case uri.host
+    when 'gist.github.com'
+      code = open("#{uri}/raw").read
+    when 'pastebin.com', 'hastebin.com'
+      code = open("#{uri.scheme}://#{uri.host}/raw#{uri.path}").read
+    else
+      logger.info "Unrecognized paste service: #{uri}"
+    end
+
+    @versions = settings.puppet_versions
+    @disabled = settings.disabled_lint_checks
+    # loads lint into global namespace, but I don't see an alternative
+    @checks   = PuppetValidator::Validators::Lint.all_checks
+    @location = location
+    @code     = sanitize_code(code)
+
+    erb :index
+  end
+
+  get '/referer' do
```

Comment:
Should this be `/load/referer` instead?

Reply:
Not sure - possibly to make it more descriptive or for some other reason?